Source/Core/VideoCommon/RenderBase.cpp

```diff
@@ -1163,30 +1146,69 @@ void Renderer::EndUIFrame()
+  if (GetConfiguredAspectMode() != AspectMode::Auto && !g_ActiveConfig.bWidescreenHack)
+    return;
+
+  // Modify the threshold based on which aspect ratio we're already using:
+  // If the game's in 4:3, it probably won't switch to anamorphic, and vice-versa.
+  static constexpr u32 TRANSITION_THRESHOLD = 3;
+
+  const auto looks_normal = [](auto& counts) {
+    return counts.normal_vertex_count > counts.anamorphic_vertex_count * TRANSITION_THRESHOLD;
+  };
+  const auto looks_anamorphic = [](auto& counts) {
+    return counts.anamorphic_vertex_count > counts.normal_vertex_count * TRANSITION_THRESHOLD;
+  };
+
+  const auto& persp = flush_statistics.perspective;
+  const auto& ortho = flush_statistics.orthographic;
+
+  const auto ortho_looks_anamorphic = looks_anamorphic(ortho);
+
+  if (looks_anamorphic(persp) || ortho_looks_anamorphic)
   {
-    m_aspect_wide = suggested == AspectMode::AnalogWide;
+    // If either perspective or orthographic projections look anamorphic, it's a safe bet.
+    m_is_game_widescreen = true;
   }
-  else if (SConfig::GetInstance().bWii)
+  else if (looks_normal(persp) || (m_was_orthographically_anamorphic && looks_normal(ortho)))
   {
-    m_aspect_wide = Config::Get(Config::SYSCONF_WIDESCREEN);
+    // Many widescreen games (or AR/GeckoCodes) use anamorphic perspective projections
+    // with NON-anamorphic orthographic projections.
+    // This can cause incorrect changes to 4:3 when perspective projections are temporarily not
+    // shown. e.g. Animal Crossing's inventory menu.
+    // Unless we were in a sitation which was orthographically anamorphic
+    // we won't consider orthographic data for changes from 16:9 to 4:3.
+    m_is_game_widescreen = false;
   }
-  else
-  {
-    // Heuristic to detect if a GameCube game is in 16:9 anamorphic widescreen mode.
-    const auto [flush_count_4_3, flush_count_anamorphic] =
-        g_vertex_manager->ResetFlushAspectRatioCount();
-    const size_t flush_total = flush_count_4_3 + flush_count_anamorphic;
 
-    // Modify the threshold based on which aspect ratio we're already using: if
-    // the game's in 4:3, it probably won't switch to anamorphic, and vice-versa.
-    if (m_aspect_wide)
-      m_aspect_wide = !(flush_count_4_3 > 0.75 * flush_total);
-    else
-      m_aspect_wide = flush_count_anamorphic > 0.75 * flush_total;
+  m_was_orthographically_anamorphic = ortho_looks_anamorphic;
+}
+
+void Renderer::Swap(u32 xfb_addr, u32 fb_width, u32 fb_stride, u32 fb_height, u64 ticks)
+{
+  if (SConfig::GetInstance().bWii)
+    m_is_game_widescreen = Config::Get(Config::SYSCONF_WIDESCREEN);
+
+  // If widescreen hack is disabled consider configured aspect ratio the game's aspect ratio.
+  // With widescreen hack enabled we only use it as the target aspect ratio,
+  // and use the widescreen heuristic (on GameCube) to adjust to this target.
+  if (!g_ActiveConfig.bWidescreenHack)
```

Comment:
Does or should the widescreen hack affect Wii games?  Should this be an 'else if' with the above?

Reply:
The widescreen hack can be relevant in Wii games.
It can be used to produce ultra-widescreen, for one.

Reply:
You can also use it to make 4:3 Wii games be 16:9 – which it seems like this PR breaks, now that I'm looking! We need to set `m_is_game_widescreen = false` when `g_ActiveConfig.aspect_mode == AspectMode::Analog`, even if `Config::Get(Config::SYSCONF_WIDESCREEN) && g_ActiveConfig.bWidescreenHack`. This is the same problem as [issue 11237](https://bugs.dolphin-emu.org/issues/11237) (which is fixed in master).

Reply:
Thanks @JosJuice .  Can you name an example of a 4:3 Wii game?  It's possible I have one but can't think of one off hand.

Reply:
N64 VC games are maybe the most well-known ones. When it comes to disc games, I can only think of less popular titles (Anubis II, Tamagotchi: Party On...), but there is a complete list here if you want it: https://github.com/dolphin-emu/dolphin/search?l=INI&q=SuggestedAspectRatio

Reply:
Thanks!  Of that list, I also have Monster Lab.  So I can help test with that too.

Reply:
@JosJuice the `suggested_aspect_mode` problem should now be fixed.